### PR TITLE
refactor(stream): migrate HashJoinExecutor to ExecutorV2

### LINKED ISSUE #1394: stream: migrate HashJoinExecutor to ExecutorV2

### PR DESCRIPTION
**Pull request by wangrunji0408**
## What's changed and what's your intention?

***PLEASE DO NOT LEAVE THIS EMPTY !!!***

As title. This PR also:
- adds `barrier_align` v2
- removes v1 `MockAsyncSource` and makes v2 `MockSource` async-able.

I'm not sure whether my migration is correct. For example the function `clear_cache` is now unused and I don't know where to call it...

## Checklist

- [x] I have written necessary docs and comments
- [x] I have added necessary unit tests and integration tests

## Refer to a related PR or issue link (optional)
resolve #1394

### CODE REVIEW

**Comment by BugenZhao** [comment]
> For example the function clear_cache is now unused and I don't know where to call it...

Exactly🤣. We may further investigate how to provide this capability in the future.

**Comment by skyzh** [comment]
clear_cache is to-be-deprecated. It won't be used in the future.

**Comment by skyzh** [comment]
Please run `./risedev check` to fix fmt 🤣

**Comment by TennyZhuang** [comment]
clear_cache is deprecated.

In the future, we will use managed cache, and for testing, we can allocate only a little memory to push all executors evict their old states.

File: src/stream/src/executor/barrier_align.rs [hunk 1]
```diff
@@ -80,9 +80,9 @@ impl BarrierAligner {
         loop {
```
Comment (BugenZhao) [inline src/stream/src/executor/barrier_align.rs:80]: Is this old `BarrierAligner` still used? IIRC, only Lookup uses it besides HashJoin. If so, we may directly remove this.

**Review by BugenZhao — APPROVED** [review]
LGTM.

File: src/stream/src/executor/barrier_align.rs (hunk 1 above)
Comment (skyzh) [reply src/stream/src/executor/barrier_align.rs:80]: Lookup now has its own barrier aligner.

Comment (wangrunji0408) [reply src/stream/src/executor/barrier_align.rs:80]: Yes, the v2 `UnionExecutor` depends on it. We can refactor it in future PR.

**Pull request merged**